### PR TITLE
Upgrade to optimization namespace change

### PR DESCRIPTION
**Pull request by michaeldiamant**
This is the ota-external analog to https://github.com/nToggle/hellgate-external/pull/37 in order to prep for merging of https://github.rp-core.com/dtech/optimization/pull/24.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ object Dependencies {
     private def dtModule(name: String, module: String, version: String) =
       s"com.rubiconproject.dt.$name" %% s"$name-$module" % version
 
-    val optimization = dtModule("optimization", "scala-with-java-conversions", "0.2.0")
+    val optimization = dtModule("optimization", "scala-with-java-conversions", "0.3.0-namespacing-SNAPSHOT")
```
Comment (michaeldiamant) [inline project/Dependencies.scala]: I will bump to non-SNAPSHOT prior to merge.

**Comment by michaeldiamant** [comment]
I assume folks will call out the concern raised in the hellgate-external PR:  https://github.com/nToggle/hellgate-external/pull/37#issuecomment-391488827.  I will address this in a forthcoming commit for ota-external (likely tomorrow AM).

**Comment by michaeldiamant** [comment]
@saharMlowell  / @leifwickland Requesting another round of review after providing the mentioned package name / layout change for consistency via https://github.com/nToggle/ota-external/pull/4/commits/4a7ce0b5fdf61ba7e5ea72a066d2b49130dce5d8.

File: src/main/protobuf/com/rubiconproject/dt/ota/impressionresult/impression_result.proto [hunk 2]
```diff
@@ -17,7 +17,7 @@ option (scalapb.options) = {
 message ImpressionResult {
     uint64 decisionReachedTimestampMs = 1;
     string auctionId = 2;
-    optimization.OptimizationAttributes attributes = 3;
+    com.rubiconproject.dt.optimization.OptimizationAttributes attributes = 3;
```
Comment (saharMlowell) [inline src/main/protobuf/com/rubiconproject/dt/ota/impressionresult/impression_result.proto]: out of curiosity, why do we add: `com.rubiconproject.dt.`? would compile with or without it.

Comment (drstevens) [reply src/main/protobuf/com/rubiconproject/dt/ota/impressionresult/impression_result.proto]: https://github.rp-core.com/dtech/optimization/pull/24/files#diff-551422096b7a66f35b66cd49881a3893R5

**Comment by michaeldiamant** [comment]
I've bumped to non-SNAPSHOT optimization version and will now merge.  I plan to address several other pending optimization PRs and reflect into ota-external before cutting an ota-external release.  If folks would rather see a different approach, let's discuss.

**Pull request merged**